studio/hooks/queries/usePermissions.ts

```diff
@@ -0,0 +1,68 @@
+import useSWR from 'swr'
+import jsonLogic from 'json-logic-js'
+import { find } from 'lodash'
+import { useRouter } from 'next/router'
+
+import { useFlag, useStore } from 'hooks'
+import { get } from 'lib/common/fetch'
+import { API_URL, IS_PLATFORM } from 'lib/constants'
+import { Organization, Project } from 'types'
+
+export function usePermissions(returning?: 'minimal') {
+  let url = `${API_URL}/profile/permissions`
+
+  if (returning) {
+    const query = new URLSearchParams({ returning }).toString()
+    url = `${url}?${query}`
+  }
+
+  const { data: data, error } = useSWR<any>(url, get, { loadingTimeout: 10000 })
+  const anyError = data?.error || error
+
+  return {
+    permissions: anyError ? undefined : data,
+    isLoading: !anyError && !data,
+    isError: !!anyError,
+  }
+}
+
+export function checkPermissions(action: string, resource: string, data?: object) {
+  if (!IS_PLATFORM) return true
+
+  const enablePermissions = useFlag('enablePermissions')
+  if (!enablePermissions) return true
+
+  const { app, ui } = useStore()
+  const router = useRouter()
+
+  let organization_id: number | undefined
+  const { ref, slug } = router.query
+  if (ref) {
+    const project = find(app.projects.list(), { ref }) as Project | undefined
+    organization_id = project?.organization_id
+  } else if (slug) {
+    const organization = find(app.projects.list(), { ref }) as Organization | undefined
+    organization_id = organization?.id
+  }
```

Comment:
Doesn't look like `organization_id` is used? I'm assuming it's here for future use?

Reply:
whoa sharp eyes!